src/server/connection.rs

```diff
@@ -1491,8 +1495,58 @@ impl Connection {
         self.video_ack_required = lr.video_ack_required;
     }
 
+    #[cfg(target_os = "windows")]
+    async fn handle_multiple_user_sessions(&mut self, lr: LoginRequest) -> bool {
+        if let Some(login_request::Union::PortForward(_)) = lr.union {
+            return true;
+        } else {
+            let active_sessions = crate::platform::get_all_active_sessions();
+            if active_sessions.len() <= 1 {
+                return true;
+            }
+            let usid;
+            match lr.option.user_session.parse::<u32>() {
+                Ok(n) => usid = Some(n),
+                Err(..) => usid = None,
+            }
+            let current_process_usid = crate::platform::get_current_process_session_id();
+            if usid.is_none() {
+                let mut res = Misc::new();
+                let mut rdp = Vec::new();
+                for session in active_sessions {
+                    let u_sid = &session[0];
+                    let u_name = &session[1];
+                    let mut rdp_session = RdpUserSession::new();
+                    rdp_session.user_session_id = u_sid.clone();
+                    rdp_session.user_name = u_name.clone();
+                    rdp.push(rdp_session);
+                }
+                res.set_rdp_user_sessions(RdpUserSessions {
+                    rdp_user_sessions: rdp,
+                    ..Default::default()
+                });
+                let mut msg_out = Message::new();
+                msg_out.set_misc(res);
+                self.send(msg_out).await;
+                return true;
+            }
+            if usid != self.user_session_id && usid != Some(current_process_usid) {
+                self.on_close("Reconnecting...", false).await;
+                std::thread::spawn(move || {
+                    let _ = ipc::connect_to_user_session(usid);
+                });
+                return false;
+            } else if usid.is_some() {
+                self.user_session_id = usid.clone();
+                return true;
+            }
+            true
+        }
+    }
+
     #[cfg(not(any(target_os = "android", target_os = "ios")))]
     fn try_start_cm_ipc(&mut self) {
+        let usid = self.user_session_id;
```

Comment:
cm is always in the console session, related to auth order changes.

Reply:
Got it. If there are any more other changes required let me know, I will finish it tonight.